plugins/locale.js

```diff
@@ -1,30 +1,38 @@
 import { join } from "node:path";
-import { existsSync, readFileSync } from "node:fs";
+import fs from "node:fs";
 import fp from "fastify-plugin";
+import PathResolver from "../lib/path.js";
 import chalk from "chalk";
-import emojiFlags from "emoji-flags";
 
-// function that takes a locale string and returns the emoji flag for that locale
-// handles formats like en-US, en, en-US-POSIX
-// map countries to languages
-function getFlag(locale) {
-  if (!locale) return "";
-  const countryCode = locale.split("-")[1];
-  if (!countryCode) return "";
-  return `${emojiFlags.countryCode(countryCode).emoji}  `;
+async function checkFileExists(filename) {
+  try {
+    await fs.promises.access(filename, fs.F_OK);
+    return true;
+  } catch {
+    return false;
+  }
 }
 
-export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "" }) {
-  const localeFilePath = join(cwd, "locale", locale) + ".json";
-  if (existsSync(localeFilePath)) {
+export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "en", development }) {
+  const resolver = new PathResolver({ development, cwd });
+
+  const contentFile = await resolver.resolve("./content");
+  const compiledFileExtension = contentFile.typescript ? ".ts" : ".mjs";
+
+  const localeFilePath = join(cwd, "locales", locale) + compiledFileExtension;
+  const localeFileExists = await checkFileExists(localeFilePath);
+
+  fastify.decorate("readTranslations", async () => {
     try {
-      const translations = JSON.parse(readFileSync(localeFilePath, { encoding: "utf8" }));
-      fastify.decorate("translations", translations);
-      fastify.log.debug(
-        `${getFlag(locale)}${chalk.magenta("translations")}: loaded file "${localeFilePath}" for locale "${locale}"`
-      );
+      const { messages } = await import(`${localeFilePath}?s=${Date.now()}`);
+      fastify.log.debug(`🌏 ${chalk.magenta("translations")}: loaded file "${localeFilePath}" for locale "${locale}"`);
+
+      return messages;
     } catch (err) {
-      fastify.log.error(`Error reading translation file: ${localeFilePath}`, err);
+      if (localeFileExists) {
+        fastify.log.error(`Error reading translation file: ${localeFilePath}`, err);
```

Comment:
Wondering if you may as well move the checkFileExists into here so that its fresh when you need it and avoids doing the work when not needed. There could probably be a case where the user deletes the locale file between when that check was made and an error being thrown here